daemon/src/model/cfd.rs

```diff
@@ -721,8 +721,8 @@ impl Cfd {
         // TODO: Taker should take this from the maker, optionally calculate and verify
         // whether they match
         let hours_to_charge = 1;
-
-        let funding_fee = FundingFee::new(self.margin(), funding_rate, hours_to_charge)?;
+        let funding_fee =
+            calculate_funding_fee(self.counterparty_margin(), funding_rate, hours_to_charge)?;
```

Comment:
Does the `counterparty_margin` for taker and `margin` for maker make sense? This would mean we always use the short margin (at a leverage of 1) to apply the funding fee. I think this makes sense for now.

Reply:
I think this is fair for now even though it is not inline with how funding fees are calculated on Bitmex (and probably others) 
 
We should think about how we can document/tell this to the user though.

Reply:
> At the moment we always use the the short margin 

Is that true? The result of `counterparty_margin` changes depending on who is calling this. It would seem that each party would be calculating a different funding fee here.

Reply:
> > At the moment we always use the the short margin
> 
> Is that true? The result of `counterparty_margin` changes depending on who is calling this. It would seem that each party would be calculating a different funding fee here.

It actually does *not* change. We keep an internal book of fees. The margins should be stable - unless I'm missing something. 
At some point we said that the margin would change, but I think that was a misunderstanding. Differently said: The payout curve calculates the payout based on the given parameters, but the given margin will always be the same, while the bookkeeping of the fees changes.

Reply:
All fair points.

In any case, I was wrong. I thought we were calling `calculate_funding_fee` with the result of `counterparty_margin` as an argument, for both the taker and the maker, which would be wrong. After checking out the PR I can see that it is not the case, so all is well.

Reply:
> > At the moment we always use the the short margin
> 
> Is that true? The result of `counterparty_margin` changes depending on who is calling this. It would seem that each party would be calculating a different funding fee here.

Very good note @luckysori ! cc @da-kami @bonomat 

Can we have a unit test for this that ensures we compute the same funding rate on both sides?

Reply:
you mean on `model::Cfd`? 
The logic of that is in `accept_rollover_proposal` which uses `self.margin()` and `handle_rollover_accepted_taker` which uses `self.counterparty_margin`. 
This is not totally correct, because this assumes that the maker is always short and the taker always long!

In order to make this clearer (and more correct) I added `model::Cfd::short_margin()` that returns the `short_margin` based on the Cfd's position. That should make it clear and easier to read. 

We should still add a unit test... but adding tests for the `Cfd` logic needs some more time because we have to build up the events. OK to tackle that in a follow up?

Reply:
> We should still add a unit test... but adding tests for the `Cfd` logic needs some more time because we have to build up the events. OK to tackle that in a follow up?

I was wondering if we can do this in a test that solely operates on `FeeAccount` etc.